src/libAtomVM/otp_socket.c

```diff
@@ -752,6 +766,7 @@ static term nif_socket_close(Context *ctx, int argc, term argv[])
     }
 
 #endif
+    rsrc_obj->buf_size = DEFAULT_BUFFER_SIZE;
```

Comment:
buf_size being only BSD, this need indeed to be in an ifdef block.

Reply:
Same as ☝🏽